frontend/app/src/connection/WebsocketConnection.tsx

```diff
@@ -378,14 +378,19 @@ export class WebsocketConnection {
   /**
    * Get the session tokens to use to initialize a WebSocket connection.
    *
-   * This method returns an array containing either one or two elements:
+   * This method returns an array containing either one or three elements:
    *   1. The first element contains an auth token to be used in environments
    *      where the parent frame of this app needs to pass down an external
    *      auth token. If no token is provided, a placeholder is used.
    *   2. The second element is the session ID to attempt to reconnect to if
    *      one is available (that is, if this websocket has disconnected and is
    *      reconnecting). On the initial connection attempt, this is unset and
-   *      the return value of this method is a singleton array.
+   *      the return value of this method is a singleton array. NOTE: the
+   *      second and third elements are either set together or not at all.
```

Comment:
nit: I am wondering whether now where the return type becomes more complex, we should refactor the return type to something like:

```typescript
interface LastSessionInfo {
  sessionId: string
  lastForwardMsgId: string
}

interface SessionTokens {
  hostAuthToken: string
  lastSessionInfo?: LastSessionInfo
} 
```

Reply:
Hm, I think we can probably punt on this at least for a bit unless we start seeing session tokens used in places that aren't the `Sec-WebSocket-Protocol` header.

A nice property of this return value currently is that the value can essentially be copied directly into the header. If we were to return a more well-defined interface, we'd clean up this function but would then have to do some work to unpack the values into the header, which I think basically balances out the extra readability we get here